notebooks/ch17/classification_multiclass.md

```diff
@@ -0,0 +1,312 @@
+    - OvR/OvA: Fit a binary classification problem for each class versus the rest of the classes; result is class that has the highest probability
+- Example walking through multiclass? sklearn's logistic regression supports OvR out of the box.
+- Multilabel: More than one classification problem (e.g. a document can be positive/negative, religion/not religion, political/apolitical, etc.)
+
+## Multiclass Classification
+
+So far we have discussed binary classification, in which our classifier determines whether an observation is part of the positive class (e.g. an email is classified as spam) or the negative class (e.g. an email is not spam). However, many data science problems involve **multiclass classification**, in which we would like to classify observations as one of several different classes. For example, we may be interested in classifying emails into folders such as Family, Friends, Work, and Promotions. To solve these types of problems, we use a new method called **one-vs-rest (OvR) classification**.
+
+### One-Vs-Rest Classification
+
+In OvR classification (also known as one-vs-all, or OvA), we decompose a multiclass classification problem into several different binary classification problems. For example, we might observe training data as shown below:
+
+
+```python
+# HIDDEN
+shapes = pd.DataFrame(
+    [[1.3, 3.6, 'triangle'], [1.6, 3.2, 'triangle'], [1.8, 3.8, 'triangle'],
+     [2.0, 1.2, 'square'], [2.2, 1.9, 'square'], [2.6, 1.4, 'square'],
+     [3.2, 2.9, 'circle'], [3.5, 2.2, 'circle'], [3.9, 2.5, 'circle']],
+    columns=['$x_1$', '$x_2$', '$y$']
+)
+```
+
+
+```python
+sns.lmplot('$x_1$', '$x_2$', data=shapes, hue='$y$', markers=['^', 's', 'o'], fit_reg=False)
+plt.xlim(1.0, 4.0)
+plt.ylim(1.0, 4.0);
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_5_0.png)
+
+
+Our goal is to build a multiclass classifier that labels observations as `triangle`, `square`, or `circle` given values for $x_1$ and $x_2$. First, we want to build a binary classifier that predicts observations as `triangle`/not `triangle`:
+
+
+```python
+plot_binary(shapes, 'triangle')
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_7_0.png)
+
+
+Similarly, we build binary classifiers for the remaining classes:
+
+
+```python
+plot_binary(shapes, 'square')
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_9_0.png)
+
+
+
+```python
+plot_binary(shapes, 'circle')
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_10_0.png)
+
+
+We know that the output of the sigmoid function in logistic regression is a probability value from 0 to 1. Then to solve our multiclass classification task, we find the probability of the positive class in each binary classifier and select the class that outputs the highest positive class probability. For example, if we have a new observation with the following values:
```

Comment:
> Then to solve our multiclass classification task

nit: I think the "Then" can be removed so the sentence starts with "To solve our multiclass..."